com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/MotionVectorVertexShaderCommon.hlsl

```diff
@@ -107,6 +107,21 @@ PackedVaryingsType MotionVectorVS(inout VaryingsType varyingsType, AttributesMes
     {
         bool hasDeformation = unity_MotionVectorsParams.x > 0.0; // Skin or morph target
 
+#if defined(HAVE_VFX_MODIFICATION)
+        // Re-construct the VFX mesh (i.e. in case of procedural output like planar primitives).
+        // And fetch element index to sample the previous element matrix.
+        AttributesElement element;
+        ZERO_INITIALIZE(AttributesElement, element);
+
+        GetMeshAndElementIndex(inputMesh, element);
+
+        // NOTE: We have to re-evaluate the attributes here, not good.
+        // TODO: organize things so that VFX mesh and attributes get computed once.
+        GetElementData(element);
```

Comment:
MotionVectorVS requires us to re-evaluate all of the VFX attributes + graph information (needless cost on performance) to support motion vectors for VFX animation + vertex stack. This happens because currently none of the work done in VertMesh to configure the mesh gets transferred here. This PR should not merge until we confirm how we do that.

Reply:
CC @sebastienlagarde for discussion - basically this code forces us to re-evaluate all of the VFX code block a second time to reconstruct AttributesElement. Ideally we just share what gets created by VertMesh, but doing this would have some design implications for the rest of the times we invoke VertMesh etc.

Reply:
I'm not sure I understand why this would be is needed? Why do we need the attributes?

Reply:
Attributes are required because we need to evaluate the vertex stack for the previous frame (to compute vertex animation motion vectors)

Reply:
But again if we just re-use the attributes from VertMesh this is not necesarry to reconstruct a second time

Reply:
Reevaluation of attributes will probably be stripped away by the compiler anyway as exactly identical expressions. But this should be only relevant for HAVE_MESH_MODIFICATION right?

There's is more a space problem here: inputMesh.positionOS should be in particle space and not in global VFX space. So TransformMeshToPreviousElement should not touch it. It should be only a matter of abstracting what is objectToWorld and previousObjectToWorld matrices.

Reply:
just to clarify, you are worrying about the re-call to GetElementData(element); ?

Reply:
Yes- I worry about evaluating the entire graph blocks a second time to re-construct the element definition (which contains attributes that control the previous frame vertex stack, as well as the index to the previous element matrix). Unfortunately it's not very easy to share what gets done in VertMesh with MotionVectorVS

Reply:
We have done a call with John about this one